receptorctl/receptorctl/socket_interface.py

```diff
@@ -34,40 +46,34 @@ def read_and_parse_json(self):
         data = json.loads(text)
         return data
 
-    def readconfig(self, ctxobj):
-        yamlfile = ctxobj["config"]
-        tlsclient = ctxobj["tls-client"]
-        if yamlfile and tlsclient:
-            with open(yamlfile, "r") as yam:
-                self.yaml = yaml.load(yam, Loader=yaml.FullLoader)
-                yam.close()
-            for i in self.yaml:
-                key = i.get("tls-client", None)
-                if key:
-                     if key["name"] == tlsclient:
-                         ctxobj["key"] = key.get("key", ctxobj["key"]) # if not in config, keep the previous value
-                         ctxobj["cert"]= key.get("cert", ctxobj["cert"])
-                         ctxobj["rootcas"] = key.get("rootcas", ctxobj["rootcas"])
-                         ctxobj["insecureskipverify"] = key.get("insecureskipverify", ctxobj["insecureskipverify"])
-                         break
+    def readconfig(self, config, tlsclient):
+        with open(config, "r") as yamlfid:
+            yamldata = yaml.load(yamlfid, Loader=yaml.FullLoader)
+            yamlfid.close()
+        for i in yamldata:
+            key = i.get("tls-client", None)
+            if key:
+                 if key["name"] == tlsclient:
+                     self.rootcas = key.get("rootcas", self.rootcas)
+                     self.key = key.get("key", self.key)
+                     self.cert = key.get("cert", self.cert)
+                     self.insecureskipverify = key.get("insecureskipverify", self.insecureskipverify)
+                     break
 
     def simple_command(self, command):
+        self.connect()
         self.writestr(f"{command}\n")
         return self.read_and_parse_json()
 
-    def connect(self, ctxobj):
-        address = ctxobj["socket"]
-        key = ctxobj["key"]
-        cert = ctxobj["cert"]
-        rootcas = ctxobj["rootcas"]
-        insecureskipverify = ctxobj["insecureskipverify"]
+    def connect(self):
         if self.socket is not None:
-            raise ValueError("Already connected")
-        m = re.compile("(tcp|tls):(//)?([a-zA-Z0-9-]+):([0-9]+)|(unix:(//)?)?([^:]+)").fullmatch(address)
+            return
```

Comment:
I think it's okay to remove the error here. User can call `.close()` first if a different connection is needed

Reply:
Yep, agreed